packages/google_maps_flutter/google_maps_flutter/example/integration_test/src/maps_inspector.dart

```diff
@@ -612,6 +627,186 @@ void runTests() {
+
+    bool cameraIdleFired = false;
+
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: GoogleMap(
+        key: key,
+        initialCameraPosition: kInitialCameraPosition,
+        onCameraIdle: () {
+          cameraIdleFired = true;
+          cameraFiredStream.add(null);
+        },
+        onMapCreated: (GoogleMapController controller) {
+          controllerCompleter.complete(controller);
+        },
+      ),
+    ));
+
+    final GoogleMapController controller = await controllerCompleter.future;
+
+    await tester.pumpAndSettle();
+    // TODO(cyanglaz): Remove this after we added `mapRendered` callback, and `mapControllerCompleter.complete(controller)` above should happen
+    // in `mapRendered`.
+    // https://github.com/flutter/flutter/issues/54758
+    await Future<void>.delayed(const Duration(seconds: 1));
+
+    // Drain any event that might have been added from the initial setup or
+    // resetting the camera position.
+    if (cameraIdleFired) {
+      await cameraFiredQueue.next;
+      cameraIdleFired = false;
+    }
+
+    final CameraUpdate cameraUpdate =
+        _getCameraUpdateForType(_cameraUpdateTypeVariants.currentValue!);
+    await controller.animateCamera(cameraUpdate);
+
+    // Check that position is not updated immediately to the target.
+    final CameraPosition beforeFinishedPosition =
+        await inspector.getCameraPosition(mapId: controller.mapId);
+    await _checkCameraUpdateByType(
+        _cameraUpdateTypeVariants.currentValue!,
+        beforeFinishedPosition,
+        null,
+        controller,
+        (Matcher matcher) => isNot(matcher));
+
+    // Check that position is animated after the animation is done.
+    expect(cameraIdleFired, isFalse);
+    await cameraFiredQueue.next;
+    expect(cameraIdleFired, isTrue);
+    final CameraPosition afterFinishedPosition =
+        await inspector.getCameraPosition(mapId: controller.mapId);
+    await _checkCameraUpdateByType(
+        _cameraUpdateTypeVariants.currentValue!,
+        afterFinishedPosition,
+        beforeFinishedPosition,
+        controller,
+        (Matcher matcher) => matcher);
+
+    await tester.pumpAndSettle();
+  }, variant: _cameraUpdateTypeVariants);
+
+  testWidgets(
+    'testAnimateCameraWithConfiguration',
+    (WidgetTester tester) async {
+      final Key key = GlobalKey();
+      final Completer<GoogleMapController> controllerCompleter =
+          Completer<GoogleMapController>();
+      final GoogleMapsInspectorPlatform inspector =
+          GoogleMapsInspectorPlatform.instance!;
+
+      final StreamController<void> cameraFiredStream = StreamController<void>();
+      final StreamQueue<void> cameraFiredQueue =
+          StreamQueue<void>(cameraFiredStream.stream);
+
+      // We set the duration to 100ms and check that the animation is completed
+      // faster than the normal animation (~300ms on Android) to make sure the
```

Comment:
This is run on other platforms as well, and it's not clear from the comment why this test is correct on those platforms.

Reply:
Comment updated